src/nnet3/nnet-component-itf.h

```diff
@@ -408,6 +403,10 @@ class UpdatableComponent: public Component {
   /// a different value than x will returned.
   BaseFloat LearningRate() const { return learning_rate_; }
 
+  /// \brief Sets is_gradient_ to true and sets learning_rate_ to 1, ignoring
+  /// learning_rate_factor_.
+  void SetAsGradient() { SetActualLearningRate(1.0); is_gradient_ = true; }
```

Comment:
SetActualLearningRate() is a virtual function that was overwritten in CompositeComponent, so we have to call it here.